pallets/delegation/src/migrations.rs

```diff
@@ -0,0 +1,480 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2021 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use sp_std::{boxed::Box, collections::btree_map::BTreeMap, vec};
+
+use crate::*;
+
+// Contains the latest version that can be upgraded.
+// For instance, if v1 of the storage is the last one available, only v0 would
+// be upgradeable to v1, so the value of LATEST_UPGRADEABLE_VERSION would be 0.
+// This needs to be bumped (by 1) every time a new storage migration should take
+// place. Along with this, a new version migrator must be defined and added to
+// the `migrations` vector or the `StorageMigrator`.
+const LATEST_UPGRADEABLE_VERSION: u16 = 0;
+
+// The trait that each version migrator must implement.
+trait VersionMigrator<T: Config> {
+	#[cfg(feature = "try-runtime")]
+	fn pre_migrate(&self) -> Result<(), DelegationMigrationError>;
+	fn migrate(&self) -> Weight;
+	#[cfg(feature = "try-runtime")]
+	fn post_migrate(&self) -> Result<(), DelegationMigrationError>;
+}
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) enum DelegationMigrationError {
+	AlreadyLatest,
+	MigrationResultInconsistent,
+}
+
+// The "manager" of the pallet's storage migrations. It contains a vector of
+// version migrations each of which corresponds to a new runtime upgrade to run,
```

Comment:
Shouldn't it be `of which each`?
```suggestion
// version migrations of which each corresponds to a new runtime upgrade to run,
```

Reply:
I think it depends whether there's a comma or not. Not entirely sure about that. But I'll have a comment refining round and will keep this in mind!